### DISCUSSION TITLE
Edytor

### DISCUSSION THREAD

**Discussion by folkien** [dev-0x7C6/fdt-viewer#2] [Q&A] [answered]
Hej a będzie też edytor czy tylko viewer? Może najlepiej jakby ten edytor potrafił wyczytać ze źródeł kernela przykłady dot. roznych wpisów i podpowiedziec je/uzyć ich przy edycji od razu?

**Comment by dev-0x7C6** [comment, marked as answer]
Cześć,

Myślałem ograniczyć się tylko do podglądu, przynajmniej w pierwszych wersjach. Ograniczając się tylko do formatu `FDT` (`dtb`), pomijam cały problem analizy leksykalnej plików `dts`. Ale może podejmę temat, bo dość fajnie pracowało mi się i analizowało jak `FDT` jest skonstruowane :)

W najbliższym czasie myślałem jeszcze dołożyć:
- poprawne wyświetlanie danych binarnych
- podstawowa obsługę z konsoli w stylu `fdt-viewer --open "sample.dtb"`
- instalacja w systemie
- ikona + integracja z środowiskiem graficznym
- ostatnio otwarte pliki
- reagowanie na zmianę otwartego pliku binarnego, np. przeładowanie kontekstu

> Może najlepiej jakby ten edytor potrafił wyczytać ze źródeł kernela przykłady dot. roznych wpisów i podpowiedziec je/uzyć ich przy edycji od razu?

Taki edytor do pracy z `devicetree` byłoby dobry jako plugin do `atoma` albo `vscode`. Wtedy praca z różnymi komponentami systemu równocześnie byłaby prostsza. Czekam aż ktoś ambitny coś zrobi w temacie, widziałem plugin do bitbake ale mi nie działał tak jak był przedstawiany.
